### ISSUE TITLE
[Accessibility] Admin Order Notes field missing status message

### ISSUE THREAD

**Issue by amberhinds** [woocommerce/woocommerce#64512]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

On a single order screen in the admin, there is a way to add notes to the order. These notes can be either private notes or notes that are emailed to the customer. Examples:

<img width="303" height="649" alt="Image" src="https://github.com/user-attachments/assets/147c547f-6663-45db-b5f3-3145e699a998" />

When an order note is added, the Order Notes meta box refreshes and the new note is visible, but nothing is communicated to screen reader users. There is no way for them to know that the note was successfully saved or emailed without exploring the page.

This is a failure of [WCAG 4.1.3 Status Messages (Level AA).](https://www.w3.org/WAI/WCAG22/Understanding/status-messages.html)

### Expected behavior

When an order note is saved, there should be a screen reader announcement communicating the successful save of the order note.

This was recently [remediated in the Product Vendor add-on](https://linear.app/a8c/issue/WOOA11Y-68/accessibility-status-message-not-provided-vendor-individual-order-view) for the product vendor view; however, it still needs to be fixed in Core Woo.

The approach used in Product Vendors was to add a visible alert that is also announced by screen readers, as shown in this screenshot.

<img width="1534" height="892" alt="Image" src="https://github.com/user-attachments/assets/a93157e6-74f3-423b-ad8a-94f1a020b6d5" />

That is a good approach. @vikrampm1 may have additional information on this fix.

It would also be sufficient to use the use the [wp.a11y.speak() JavaScript method](https://wpaccessibility.org/docs/topics/code/dynamic-changes/#using-aria-live-in-wordpress ) to add screen reader messages in the existing alert or status containers in core. 

Ultimately, the expected behavior is that there is a screen reader announcement comunicating the status of the note form submission.

### Actual behavior

Currently, there is no announcement made by screen readers when an order note is added. See this recording with VoiceOver active.

https://github.com/user-attachments/assets/4b117e22-95d5-44a1-8703-0c85f8222e8b

### Steps to reproduce

1. Go to the orders screen in WooCommerce (wp-admin) and click on any existing order.
2. Turn on a screen reader such as NVDA or VoiceOver.
3. Navigate to the Order Notes meta box.
4. Type any text in the Add note field, then click the Add button.
5. Observe that the screen reader does not announce anything, though the added note is inserted into the metabox.

### WordPress Environment

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 10.7.0
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.9.3
Log Directory Writable: ✔
WP Version: ❌ 6.8.3 - There is a newer version of WordPress available (6.9.4)
WP Multisite: –
WP Memory Limit: 128 MB
WP Debug Mode: –
WP Cron: ✔
WP Environment Type: production
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.65 (Ubuntu) mod_fcgid/2.3.9 OpenSSL/3.0.13
Server Architecture: Linux 6.8.0-71-generic x86_64
PHP Version: 8.2.29
PHP Post Max Size: 256 MB
PHP Time Limit: 120
PHP Max Input Vars: 5000
cURL Version: 8.5.0
OpenSSL/3.0.13

SUHOSIN Installed: –
MySQL Version: 10.11.13-MariaDB-ubu2004
Max Upload Size: 256 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 12
nav_menu_item: 5
page: 11
post: 2
product: 23
product_variation: 61
revision: 6
shop_order: 46
wp_font_face: 36
wp_font_family: 12
wp_navigation: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (13) ###

Accessibility Checker: by Equalize Digital – 1.40.0
AutomateWoo - Birthdays Add-on: by WooCommerce – 1.3.58
AutomateWoo - Refer A Friend Add-on: by WooCommerce – 2.8.16
AutomateWoo: by WooCommerce – 6.3.1
Classic Editor: by WordPress Contributors – 1.6.7
Pinterest for WooCommerce: by WooCommerce – 1.4.26
User Switching: by John Blackbourn – 1.11.2
WooCommerce Smooth Generator: by Automattic – 1.2.2
WooCommerce.com Update Manager: by Automattic – 1.0.3
WooCommerce Payfast Gateway: by WooCommerce – 1.7.6
WooCommerce Product Vendors: by WooCommerce – 2.5.1
WooCommerce: by Automattic – 10.7.0
WP Mail Logging: by WP Mail Logging Team – 1.16.0

### Inactive Plugins (2) ###

Braintree for WooCommerce Payment Gateway: by WooCommerce – 3.10.0
Max Mega Menu: by megamenu.com – 3.9.2.1

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: ZAR (R)
Currency Position: left
Thousand Separator: 
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: –
Enabled Features: analytics
marketplace
order_attribution
site_visibility_badge
remote_logging
email_improvements
blueprint
point_of_sale


### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 1 KB

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/ -  Contains the woocommerce/cart block
Checkout: #8 - /checkout/ -  Contains the woocommerce/checkout block
My account: #9 - /my-account/ -  Contains the [woocommerce_my_account] shortcode
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.6.1 (update to version 4.6.2 is available)
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
Theme type: Classic theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
analytics-scheduled-import
product-block-editor
experimental-iapi-mini-cart
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: product-data-views
experimental-blocks
experimental-iapi-runtime
coming-soon-newsletter-template
minified-js
product-pre-publish-modal
products-catalog-api
settings
async-product-editor-category-field
product-editor-template-system
use-wp-horizon
rest-api-v4

Daily Cron: ❌ Not scheduled
Options: ✔
Notes: 102
Onboarding: completed

### Action Scheduler ###

Canceled: 3
Oldest: 2026-03-31 07:40:35 +0000
Newest: 2026-04-04 12:45:44 +0000

Complete: 2,306
Oldest: 2026-03-31 07:40:35 +0000
Newest: 2026-04-30 18:15:23 +0000

Failed: 8
Oldest: 2025-08-15 15:46:04 +0000
Newest: 2026-04-30 17:44:53 +0000

Pending: 22
Oldest: 2026-04-30 18:15:38 +0000
Newest: 2026-05-12 05:06:56 +0000


### Status report information ###

Generated at: 2026-04-30 18:15:50 +00:00
```
</details>

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by jamesckemp** [comment]
Thanks for the report! We'll look into it and keep you updated here.
